🚧[Work In Progress] Basic Dancho

the-adventurist opened the pull request.

karastoyanov committed:
Test Dancho

karastoyanov committed:
Git Ignore File Added

karastoyanov committed:
.gitignore update

KingTheSim committed:
I removed the 'sapi5' from line 8. sapi5 is used to call a specific module. But this module is for Windows only. By removing the script, we simply use the module that's best suited for the current system and ensure that users from other platforms don't run straight into an error upon starting the program.

karastoyanov committed:
Exercise Functions Uploaded

the-adventurist committed:
Correction of the speed of the voice. The assistant tells the weather forecast.

the-adventurist committed:
Correction of the speed of the voice. The assistant tells the weather forecast.

the-adventurist committed:
Correction of the speed of the voice. The assistant tells the weather forecast.

karastoyanov committed:
Branch restored

karastoyanov committed:
Dancho VA Restored

karastoyanov committed:
Test commit

   - this is just a test commit
   - it's containing just a simple Hello world print.

karastoyanov commented on test_dancho.py:
```diff
@@ -30,6 +30,8 @@ def say_date():
     """"Type your code here"""
     pass
 
+print("Hello world!")
```
djgjhfdg

KingTheSim committed:
say_date() code written

the-adventurist committed:
Dancho says the current time in 24 hours format.

the-adventurist committed:
Merge remote-tracking branch 'origin/development' into development

# Conflicts:
#	test_dancho.py

the-adventurist committed:
Dancho says the current time in 12 hours format. Time module has been used.

karastoyanov committed:
Project Update

karastoyanov commented on test_dancho.py:
```diff
@@ -27,11 +28,16 @@ def say_time():
 
-print("Hello world!")
+    if not spec_dates.issuperset(cur_date[:-5]):
+        speak(f"Today's {cur_date}. It's just a regular day")
```
On this line, use the `return` statement  to send the string as as result from the function `say_date` and call the function on the last line

karastoyanov commented on test_dancho.py:
```diff
@@ -27,11 +28,16 @@ def say_time():
+    if not spec_dates.issuperset(cur_date[:-5]):
+        speak(f"Today's {cur_date}. It's just a regular day")
+    else:
+        speak(f"Today's {cur_date}. {spec_dates_prompts[cur_date[:-5]]}")
```
On this line, use the `return` statement  to send the string as as result from the function `say_date` and call the function on the last line

karastoyanov committed:
Speech Recognition implementation

karastoyanov committed:
Basic Feedback from Dancho

karastoyanov committed:
Simple Voice Command Functionality

karastoyanov committed:
Testing commit

karastoyanov committed:
Greeting Function Implemented

karastoyanov committed:
Datetime minor fix, month pronuntation

karastoyanov committed:
Wikipedia Implementation

the-adventurist committed:
weather_forecast folder

the-adventurist committed:
without test_dancho

karastoyanov committed:
Merge pull request #9 from Stake-And-Rope/weather_forecast

🚧[Work In Progress] Weather Forecast

karastoyanov commented on weather_forecast/weather_forecast.py:
```diff
@@ -0,0 +1,50 @@
+import datetime
+import pyaudio
+import speech_recognition as sr
+
```
sdasds

karastoyanov replied:
Remove imported libraries `pyttxs3` 2nd `datetime` `pyaudio` and `speech_recognition`

karastoyanov commented on weather_forecast/weather_forecast.py:
```diff
@@ -0,0 +1,50 @@
+import pyaudio
+import speech_recognition as sr
+
+MASTER = input("Enter your name: ")
```
Remove from file the entire functionality for Dancho VA.

karastoyanov commented on weather_forecast/weather_forecast.py:
```diff
@@ -0,0 +1,50 @@
+day = (str(day).split(" "))[0].split("-")[2]
+weather_type = response['weather'][0]['main']
+wind_speed = response['wind']['speed']
+
```
This entire block of code should be inside a separate function (from line 20 to line 29)

The function should return string with the current weather forecast

karastoyanov reviewed (changes requested):
Weather Forecast Code Review

karastoyanov committed:
Wikipedia Search Update

karastoyanov committed:
Wikipedia import fixed

karastoyanov committed:
Code Refractor

the-adventurist committed:
Completed function retrieve data about the weather forecast.

the-adventurist commented:
I wrote the file containing the function 'retrieve_the_data_of_the_weather_forecast'. The information retrieves from the internet in JSON format. The raw JSON file contains some other information about the city and weather features that could be enclosed additionally in the returned text. I think the most important information is included.

karastoyanov committed:
Wikipedia functionality updated

karastoyanov commented on weather_forecast/weather_forecast.py:
```diff
@@ -1,50 +1,86 @@
+        }
+
+    wind_direction = [direction for direction, degrees in wind_compass_rose.items() if wind_degrees in degrees]
+
```
Try to re-format the entire string below as one. Right now the voice assistant is reading back-slashes as well.

Also, you can reduce a little bit the information which the function returns. You can try to remove the following lines:

```python
   f'Atmospheric pressure will be {pressure_hpa} hpa.' \
   f'Humidity will be {relative_air_humidity_in_percent} percent.' \
   f'The wind will be {wind_direction[0]} with a speed of {wind_speed:.1f} kilometers' \
```

karastoyanov reviewed (changes requested):
Some minor changes for the weather forecast function need to be done. Overall it's ok.

karastoyanov committed:
Weather Forecast Implementation

the-adventurist committed:
A lighter version of the function retrieve_data_about_weather_forecast.

the-adventurist commented:
I've removed the backslashes of the returned string and also have removed the information about humidity, pressure, and wind conditions. I've got rid of redundant variables also.

karastoyanov committed:
Weather Forecast Function Update

karastoyanov committed:
Weather Forecast Fix

karastoyanov commented on weather/weather_forecast.py:
```diff
@@ -19,68 +18,21 @@ def retrieve_data_about_the_weather_forecast():
-    BASE_URL = "http://api.openweathermap.org/data/2.5/weather?"
-    CITY = "Sofia"
+    base_url = "http://api.openweathermap.org/data/2.5/weather?"
+    city = "Sofia"
```
We should make this variable to be dynamic and Dancho to read the weather based on the user input.

karastoyanov replied:
Fixed with https://github.com/Stake-And-Rope/dancho-voice-assistant/commit/44c51aee3a67b83a0086dce4c0cf4081f91466d4

karastoyanov commented on weather/weather_forecast.py:
```diff
@@ -19,68 +18,21 @@ def retrieve_data_about_the_weather_forecast():
+    city = "Sofia"
 
-    url = BASE_URL + "q=" + CITY + "&appid=" + "f4e5b648421e4d481828851a5763cbf1"
+    url = base_url + "q=" + city + "&appid=" + "f4e5b648421e4d481828851a5763cbf1"
```
Try to implement `try - except` statement IF city not found --> Dancho should inform the user

karastoyanov replied:
Fixed with 44c51aee3a67b83a0086dce4c0cf4081f91466d4

karastoyanov reviewed (changes requested):
Weather Forecast functionality needs to be optimized

karastoyanov committed:
Wikipedia function update - returns URL address

karastoyanov committed:
Wikipedia browser open

the-adventurist committed:
Enhanced function with flexibility to check weather at different locations.

karastoyanov committed:
weather forecast functionality update

 - dynamically receiving the city input - IF city not found, Dancho is asking for new input
 - speed reduced for weather forecast functionality

Co-Authored-By: the-adventurist <122741712+the-adventurist@users.noreply.github.com>

karastoyanov commented on weather/weather_forecast.py:
```diff
@@ -23,20 +23,37 @@ def retrieve_data_about_the_weather_forecast(city_):
+
+    # weather_forecast_text = f'Today will be {weather_type} with {weather_type_further_description}. The current temperature is {current_temperature_in_celsius:.1f} degrees celsius and will feel like {feels_like_in_celsius:.1f} degrees celsius. The minimal temperature will be {temperature_min_celsius:.1f} degrees celsius. The maximal temperature will be {temperature_max_celsius:.1f} degrees cels ...
+    # print(weather_forecast_text)
+    # return weather_forecast_text
```
The comments above should be deleted. This block of code is obsolete and not used anymore

the-adventurist replied:
44c51aee3a67b83a0086dce4c0cf4081f91466d4 I got rid of redundant comments.

karastoyanov reviewed (changes requested):
Code review on the last commit
@the-adventurist , please fix the issue from the code review and commit the code

the-adventurist committed:
Removed redundant comments

karastoyanov committed:
pycurl whl packages upload

karastoyanov committed:
youtube folder added

karastoyanov committed:
youtube folder added

the-adventurist committed:
testing on some functionality from youtube

karastoyanov committed:
Youtube functionality updated:
 - file renamed for better visibility
 - function accepts one argument -> search, the actual string which will be passed to youtube as search word
 - .env file implemented with python-decouple library, storing the google token locally`:wq

karastoyanov committed:
dependencies update -> python-decouple

karastoyanov commented:
@Stunxzz @the-adventurist check the [f1c2623](https://github.com/Stake-And-Rope/dancho-voice-assistant/pull/3/commits/f1c262349f2457d2dd9e6ac721f1838d59f86d10) commit, I have added one new dependency to the project `python-decouple` which you will need to read the locally stored YOUTUBE_TOKEN.

the-adventurist committed:
added function about retriving ulrs at testing_youtube_functionality.py

karastoyanov committed:
google token update

karastoyanov committed:
youtube_api update

karastoyanov committed:
Spotify Functionality Added - Update .env file

karastoyanov committed:
dependencies updated

karastoyanov committed:
spotify play added

karastoyanov committed:
Spotify play track implemented

karastoyanov committed:
Spotify functionality updated

karastoyanov committed:
Spotify drafts functions ready
